File: chrome-extension/assets/icons/generate-icons.js
/**
 * Script to generate PNG icons from SVG for Chrome extension
 * Requires: sharp package (npm install sharp --save-dev)
 * 
 * Run: node generate-icons.js
 */

const fs = require('fs');
const path = require('path');

// Check if sharp is available
let sharp;
try {
  sharp = require('sharp');
} catch (e) {
  console.error('Error: sharp package is required. Install it with:');
  console.error('  npm install sharp --save-dev');
  console.error('\nOr use an online SVG to PNG converter like:');
  console.error('  https://convertio.co/svg-png/');
  console.error('  https://cloudconvert.com/svg-to-png');
  console.error('\nGenerate PNGs at: 16x16, 48x48, and 128x128 pixels');
  process.exit(1);
}

const svgPath = path.join(__dirname, 'icon.svg');
const sizes = [16, 48, 128];

async function generateIcons() {
  try {
    const svgBuffer = fs.readFileSync(svgPath);
    
    for (const size of sizes) {
      const outputPath = path.join(__dirname, `icon${size}.png`);
      
      await sharp(svgBuffer)
        .resize(size, size, {
          kernel: sharp.kernel.lanczos3
        })
        .png()
        .toFile(outputPath);
      
      console.log(`✅ Generated icon${size}.png (${size}x${size})`);
    }
    
    console.log('\n✅ All icons generated successfully!');
  } catch (error) {
    console.error('Error generating icons:', error);
    process.exit(1);
  }
}

generateIcons();

File: chrome-extension/assets/icons/generate-icons.py
#!/usr/bin/env python3
"""
Generate PNG icons from SVG for Chrome extension
Requires: pip install cairosvg pillow

Usage: python3 generate-icons.py
"""

import os
import sys

try:
    import cairosvg
except ImportError:
    print("Error: cairosvg is required. Install it with:")
    print("  pip install cairosvg")
    print("\nAlternatively, use the HTML exporter (export-icons.html)")
    print("or an online converter like https://convertio.co/svg-png/")
    sys.exit(1)

def generate_icons():
    svg_path = os.path.join(os.path.dirname(__file__), 'icon.svg')
    sizes = [16, 48, 128]
    
    if not os.path.exists(svg_path):
        print(f"Error: {svg_path} not found")
        sys.exit(1)
    
    for size in sizes:
        output_path = os.path.join(os.path.dirname(__file__), f'icon{size}.png')
        
        try:
            cairosvg.svg2png(
                url=svg_path,
                write_to=output_path,
                output_width=size,
                output_height=size
            )
            print(f"✅ Generated icon{size}.png ({size}x{size})")
        except Exception as e:
            print(f"❌ Error generating icon{size}.png: {e}")
            sys.exit(1)
    
    print("\n✅ All icons generated successfully!")

if __name__ == '__main__':
    generate_icons()



